components/StyledInputTags.js

```diff
@@ -1,47 +1,273 @@
+  &:hover {
+    color: ${props => props.theme.colors.blue[600]};
+  }
+`;
+
+const DeleteTagButton = styled(TagActionButton)`
+  color: ${props => props.theme.colors.black[400]};
+  &:hover {
+    color: ${props => props.theme.colors.black[600]};
+  }
+`;
 
 const messages = defineMessages({
   placeholder: {
     id: 'StyledInputTags.Placeholder',
-    defaultMessage: 'Add new tags by pressing Enter',
+    defaultMessage: 'Create New Tag',
+  },
+  editLabel: {
+    id: 'StyledInputTags.EditLabel',
+    defaultMessage: 'Edit Tags',
+  },
+  addLabel: {
+    id: 'StyledInputTags.AddLabel',
+    defaultMessage: 'Add Tags',
   },
 });
 
 const getOptions = tags => {
   if (!tags || !tags.length) {
-    return undefined;
+    return [];
   } else {
     return tags.map(tag => ({ label: tag, value: tag }));
   }
 };
 
-const StyledInputTags = ({ suggestedTags, value, defaultValue, ...props }) => {
+const StyledInputTags = ({ suggestedTags, value, onChange, renderUpdatedTags, defaultValue, ...props }) => {
   const { formatMessage } = useIntl();
-  const options = getOptions(suggestedTags);
+  const inputRef = React.useRef();
+  const wrapperRef = React.useRef();
+  const scrollerRef = React.useRef();
+  const [isOpen, setOpen] = React.useState(false);
+  const [tags, setTags] = React.useState(getOptions(value || defaultValue));
+
+  const handleClose = () => {
+    onChange(tags);
+    setOpen(false);
+  };
+  const handleOutsideClick = outside => {
+    if (outside) {
+      handleClose();
+    }
+  };
+  const handleToggleInput = () => {
+    isOpen ? handleClose() : setOpen(true);
+  };
+  const addTag = tag => setTags(uniqBy([...tags, { label: tag.toUpperCase(), value: tag.toUpperCase() }], 'value'));
+  const removeTag = tag => setTags(tags.filter(v => v.value !== tag));
+  const handleNewValue = e => {
+    if (e.key === 'Enter') {
+      if (e.target.value.length) {
+        e.preventDefault();
+        addTag(e.target.value);
+        e.target.value = null;
+        // Wait until new tag renders, otherwise we'll scroll to the second-last tag
+        requestAnimationFrame(() => scrollerRef.current?.scrollTo(0, Number.MAX_SAFE_INTEGER), 30);
+      } else {
+        handleClose();
+      }
+    }
+  };
+
+  useGlobalBlur(wrapperRef, handleOutsideClick);
+
   return (
-    <StyledSelectCreatable
-      isMulti
-      data-cy="styled-input-tags"
-      options={options}
-      value={getOptions(value)}
-      hideDropdownIndicator={!options}
-      hideMenu={!options}
-      placeholder={formatMessage(messages.placeholder)}
-      defaultValue={getOptions(defaultValue)}
-      {...props}
-    />
+    <Manager>
+      <Flex ref={wrapperRef} flexWrap="wrap">
+        {(renderUpdatedTags ? tags : getOptions(value))
+          .map(t => t.value)
+          ?.map(tag => (
+            <StyledTag key={tag} variant="rounded-right" mr="4px" mb="4px">
+              {tag}
+            </StyledTag>
+          ))}
```

Comment:
**[:sparkles: **Enhancement**]** Iterating the list two times doesn't seems necessary. The `.map?` is also not necessary because the first `map` doesn't have it (or both should have it). 

If it's about aliasing, we could do:

```suggestion
          .map(({value: tag}) => (
            <StyledTag key={tag} variant="rounded-right" mr="4px" mb="4px">
              {tag}
            </StyledTag>
          ))}
```

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>